code/datums/components/chem_booster.dm

```diff
@@ -0,0 +1,210 @@
+		on_off()
+	resource_storage_current -= resource_drain_amount
+
+/**
+ *	Opens the radial menu with everything
+ */
+/datum/component/chem_booster/proc/configure(datum/source)
+	SIGNAL_HANDLER_DOES_SLEEP
+	var/mob/living/carbon/human/H = wearer
+
+	var/list/radial_options = list(
+		"on_off" = image(icon = 'icons/mob/radial.dmi', icon_state = "cboost_on_off"),
+		"connect" = image(icon = 'icons/mob/radial.dmi', icon_state = "cboost_drain"),
+		"boost_1" = image(icon = 'icons/mob/radial.dmi', icon_state = "cboost_t1"),
+		"boost_2" = image(icon = 'icons/mob/radial.dmi', icon_state = "cboost_t2"),
+		)
+
+	var/choice = show_radial_menu(H, H, radial_options, null, 48, null, TRUE)
+	switch(choice)
+		if("on_off")
+			on_off()
+		if("connect")
+			connect_weapon()
+		if("boost_1")
+			update_boost(boost_tier1)
+		if("boost_2")
+			update_boost(boost_tier2)
+
+/datum/component/chem_booster/proc/on_off(datum/source)
+	if(boost_on)
+		STOP_PROCESSING(SSobj, src)
+		UnregisterSignal(wearer.reagents, COMSIG_REAGENT_ADD)
+		UnregisterSignal(wearer, COMSIG_MOB_DEATH, .proc/on_off)
+		update_boost(0, FALSE)
+		boost_on = FALSE
+		to_chat(wearer, "<span class='warning'>Halting reagent injection.</span>")
+		COOLDOWN_START(src, chemboost_activation_cooldown, 10 SECONDS)
+		return
+
+	if(COOLDOWN_CHECK(src, chemboost_activation_cooldown))
+		to_chat(wearer, "<span class='warning'>Your body is still strained after the last exposure. You need to wait a bit... unless you want to burst for excessing use.</span>")
+		return
+
+	if(resource_storage_current < resource_drain_amount)
+		to_chat(wearer, "<span class='warning'>Not enough resource to sustain operation.</span>")
+		return
+	boost_on = TRUE
+	processing_start = world.time
+	START_PROCESSING(SSobj, src)
+	RegisterSignal(wearer.reagents, COMSIG_REAGENT_ADD, .proc/late_add_chem)
+	RegisterSignal(wearer, COMSIG_MOB_DEATH, .proc/on_off)
+	update_boost(boost_amount*2, FALSE)
+	to_chat(wearer, "Commensing reagent injection.")
+
+//Updates the boost amount of the suit and effect_str of reagents if component is on. "amount" is the final level you want to set the boost to.
+/datum/component/chem_booster/proc/update_boost(amount, update_boost_amount = TRUE)
+	amount -= boost_amount
+	if(update_boost_amount)
+		boost_amount += amount
+	message_admins("effect_str modification amount > [amount]")
+	resource_drain_amount = 1
+	if(boost_on)
+		for(var/datum/reagent/R in wearer.reagents.reagent_list)
+			R.effect_str += amount
+
+//Updates the effect_str of chems that enter the body while the component is on
+/datum/component/chem_booster/proc/late_add_chem(datum/source, datum/reagent/added_chem, amount)
+	SIGNAL_HANDLER
+	for(var/datum/reagent/R in wearer.reagents.reagent_list)
+		if(!istype(R, added_chem))
+			continue
+		message_admins("added chem boosted by > [boost_amount]")
+		R.effect_str += boost_amount
+		break
+
+/datum/component/chem_booster/proc/connect_weapon()
```

Comment:
Reminder new procs need to be autodocced

Reply:
changed